data/src/main/java/io/gnosis/data/repositories/TransactionRepository.kt

```diff
@@ -33,18 +32,47 @@ class TransactionRepository(
     suspend fun submitConfirmation(safeTxHash: String, signedSafeTxHash: String): TransactionDetails =
         gatewayApi.submitConfirmation(safeTxHash, TransactionConfirmationRequest(signedSafeTxHash))
 
+    suspend fun proposeTransaction(
+        safeAddress: Solidity.Address,
+        value: BigInteger = BigInteger.ZERO,
+        data: String? = null,
+        nonce: BigInteger,
+        operation: Operation = Operation.CALL,
+        safeTxGas: Long = 0,
+        baseGas: Long = 0,
+        gasPrice: Long = 0,
+        gasToken: Solidity.Address = "0x00".asEthereumAddress()!!,
+        refundReceiver: Solidity.Address? = null,
+        safeTxHash: String,
+        sender: Solidity.Address,
+        signature: String,
+        origin: Solidity.Address? = null
+    ): Response<Unit> =
```

Comment:
I think that you don't need a return value for responses that don't have a body. Unless, you are doing something with the http code.

Reply:
I tried. It doesn't work without a return value. Let me have another look.

Reply:
I was mistaken. It works without a return type. So I removed it.